### PR TITLE
gcc@8 8.2.0

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

**Comment by iMichka** [comment]
```
strip: unable to copy file '/home/linuxbrew/.linuxbrew/Cellar/gcc@8/8.2.0/lib/libgcc_s.so.1'; reason: Permission denied
strip: /home/linuxbrew/.linuxbrew/Cellar/gcc@8/8.2.0/lib/libgcc_s.so.1: cannot set time: Operation not permitted
strip: unable to copy file '/home/linuxbrew/.linuxbrew/Cellar/gcc@8/8.2.0/lib/libstdc++.so.6'; reason: Permission denied
strip: /home/linuxbrew/.linuxbrew/Cellar/gcc@8/8.2.0/lib/libstdc++.so.6: cannot set time: Operation not permitted
```

**Comment by xu-cheng** [comment]
FYI, you probably should also include this patch https://github.com/Homebrew/homebrew-core/pull/30639/

**Comment by xu-cheng** [comment]
> strip: unable to copy file '/home/linuxbrew/.linuxbrew/Cellar/gcc@8/8.2.0/lib/libgcc_s.so.1'; reason: Permission denied
strip: /home/linuxbrew/.linuxbrew/Cellar/gcc@8/8.2.0/lib/libgcc_s.so.1: cannot set time: Operation not permitted
strip: unable to copy file '/home/linuxbrew/.linuxbrew/Cellar/gcc@8/8.2.0/lib/libstdc++.so.6'; reason: Permission denied
strip: /home/linuxbrew/.linuxbrew/Cellar/gcc@8/8.2.0/lib/libstdc++.so.6: cannot set time: Operation not permitted

I think this is because these files now are set to `(0444/-r--r--r--)` mode.
So the fix would be:
```ruby
Pathname.glob(prefix/"**/*").each do |f|
  if f.file? && (f.elf? || f.extname == ".a")
    f.ensure_writable { system "strip", "--strip-unneeded", "--preserve-dates", f }
  end
end
```

File: Formula/gcc@8.rb [hunk 1]
```diff
@@ -284,3 +288,17 @@ def post_install
     assert_equal "Done\n", `./test`
   end
 end
+
+__END__
+diff --git a/gcc/graphite.h b/gcc/graphite.h
+index 4e0e58c..be0a22b 100644
+--- a/gcc@8/graphite.h
++++ b/gcc@8/graphite.h
```
Comment (xu-cheng) [inline Formula/gcc@8.rb]: `s/gcc@8/gcc/`

**Comment by iMichka** [comment]
```
==> Patching
patching file gcc/graphite.h
Error: undefined method `ErrorDuringExecution' for Utils:Module
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/popen.rb:18:in `safe_popen_write'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/patch.rb:70:in `apply'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1104:in `each'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1104:in `patch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/build.rb:130:in `block (2 levels) in install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1117:in `block in brew'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2037:in `block (2 levels) in stage'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils.rb:564:in `with_env'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2036:in `block in stage'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:120:in `block in unpack'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:189:in `block in mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:55:in `block in run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:55:in `chdir'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:55:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:188:in `mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:115:in `unpack'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:93:in `stage'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/forwardable.rb:202:in `stage'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2014:in `stage'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1112:in `brew'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/build.rb:124:in `block in install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils.rb:564:in `with_env'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/build.rb:121:in `install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/build.rb:202:in `<main>'
```

**Comment by xu-cheng** [comment]
So, the gcc 8 now creates two symlinks `lib/"libgcc_s.so.1", lib/"libstdc++.so.6"`, which are in conflicted with the gcc formula. So I pushed a commit to remove them.

@iMichka Please review my additional changes when you are free. 😉

**Comment by iMichka** [comment]
Thanks @xu-cheng !

**Pull request closed without merging**